noodles-editor/src/noodles/components/op-components.tsx

```diff
@@ -512,32 +513,33 @@ function NodeHeader({ id, type, op }: { id: string; type: OpType; op: OperatorIn
 
       const newQualifiedId = generateQualifiedPath(trimmedName, op.containerId)
       const isContainer = type === 'ContainerOp'
+      const store = getOpStore()
 
       // Update the operator itself
-      opMap.set(newQualifiedId, op)
+      store.setOp(newQualifiedId, op)
       op.id = newQualifiedId
 
       // If this is a container, update all children nodes and their operators
       if (isContainer) {
-        const childOps = Array.from(opMap.values()).filter(childOp =>
+        const childOps = store.getAllOps().filter(childOp =>
           childOp.id.startsWith(`${id}/`)
         )
 
         for (const childOp of childOps) {
           const oldChildId = childOp.id
           // Replace only the exact container path at the start
           const newChildId = newQualifiedId + oldChildId.slice(id.length)
-          opMap.set(newChildId, childOp)
+          store.setOp(newChildId, childOp)
           childOp.id = newChildId
           // TODO: this is a hack. We should hook into some sort of "after update" event
-          queueMicrotask(() => opMap.delete(oldChildId))
+          queueMicrotask(() => store.deleteOp(oldChildId))
         }
       }
 
       // Give React time to update the component tree before deleting the old id
       // TODO: this is a hack. We should hook into some sort of "after update" event
       queueMicrotask(() => {
-        opMap.delete(id)
+        store.deleteOp(id)
       })
```

Comment:
I wonder if this is still needed? Note to look back at what issue this micro task solved